### PR TITLE
Updating csi-provisioner and csi-node-driver

### PR DESCRIPTION
**Pull request by Maleware**
# Description

As of the update for the mirror action https://github.com/stackabletech/docker-images/pull/738 path are going to change in order to automate upload of dependencies to Nexus and Harbor

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: deploy/helm/secret-operator/values.yaml
Comment (razvan) [inline deploy/helm/secret-operator/values.yaml:11]: This is a new major version. Did you research if it's backwards compatible with the previous one?

Comment (lfrancke) [reply deploy/helm/secret-operator/values.yaml:11]: https://github.com/kubernetes-csi/external-provisioner/blob/v5.0.1/CHANGELOG/CHANGELOG-5.0.md

> The external-provisioner now needs permissions to patch persistentvolumes. Please update your RBACs appropriately.

I have no idea if we use/need that. But it does not seem like a straightforward upgrade at least.

**Comment by Maleware** [comment]
Updating the secret operator accordingly https://github.com/stackabletech/secret-operator/pull/479

File: deploy/helm/secret-operator/values.yaml
Comment (nightkr) [reply deploy/helm/secret-operator/values.yaml:23]: Looking at the PR it came from (https://github.com/kubernetes-csi/external-provisioner/pull/1155): yes we do need the permission, no it doesn't look like we have take the permission already.

Comment (Maleware) [reply deploy/helm/secret-operator/values.yaml:23]: Yes, I'm updating it atm. I made another PR which was not the smartest thing. Put those together in this one now

**Comment by Maleware** [comment]
Integration test: https://testing.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/secret-operator-it-custom/3/

**Comment by Maleware** [comment]
Tests failed on hyperscaler, where all right locally:

```
--- PASS: kuttl (121.21s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/kerberos_krb5-1.21.1_openshift-false (62.93s)
        --- PASS: kuttl/harness/tls_openshift-false (23.70s)
        --- PASS: kuttl/harness/listener_openshift-false (34.55s)
PASS

```

**Comment by Maleware** [comment]
We waiting on the Tests for OpenShift before we merge it. Going to be done by @adwk67

**Comment by adwk67** [comment]
:green_circle: `secret-operator` tests on OKD with helm:
```
--- PASS: kuttl (121.91s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/tls_openshift-true (70.74s)
        --- PASS: kuttl/harness/listener_openshift-true (84.08s)
        --- PASS: kuttl/harness/kerberos_krb5-1.21.1_openshift-true (42.06s)
PASS
```
:green_circle: `secret-operator` tests on OKD with OLM bundles and updated CSDI versions:
```
--- PASS: kuttl (95.14s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/kerberos_krb5-1.21.1_openshift-true (29.02s)
        --- PASS: kuttl/harness/tls_openshift-true (52.01s)
        --- PASS: kuttl/harness/listener_openshift-true (55.50s)
PASS
```
:green_circle: `hdfs-operator` test on OKD with OLM bundles and updated CSDI versions:
```
--- PASS: kuttl (1013.60s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/kerberos_hadoop-latest-3.3.6_zookeeper-latest-3.9.2_krb5-1.21.1_kerberos-realm-PROD.MYCORP_kerberos-backend-mit_openshift-true (1003.10s)
PASS
```

**Review by adwk67 — APPROVED** [review]

**Pull request merged**
